### DISCUSSION TITLE
Monthly update and paid feature plan

### DISCUSSION THREAD

**Discussion by izhangzhihao** [izhangzhihao/intellij-rainbow-brackets#2586] [General]
Hi, all paid users,

 Firstly, thanks for your support! That means a lot to me. And for those who contributed to this project before by OpenCollective or code or other ways, please contact me and I will give you a free license.

This project is **five years old**, and I have maintained this project for five years. Over the past year, I didn't spend lots of time on this project because I've been busy with work, and there have been fewer updates. But I still keep doing bug fixes and small feature requests.  

Please believe that I have put unparalleled energy into this project. Feature requests and bug fixes are not done without any effort. For the sustainable development of this project, I will make some parts of this project (the C# part is completely rewritten) and the new ones added in the future as paid features.

**And, the open-source part of this project will still stay free, and free for always.**

Now I'm trying to start an "At Least One Feature Request or Bug Fix Per Month Plan", which means that I will at least make one feature request or bug fix done every month for paid users, and this plugin will update monthly. This basically improves the value of this plugin for paid users and makes this plugin live longer and better.

Please just fire your feature request or bug report, and I will try to fix them. And don't forget to upload your screenshots of "Help -> Register or Manage Licenses". And if someone else already fired a feature request or bug report, please 👍 for that.

Happy coding~

**Comment by ThexXTURBOXx** [comment]
I am sorry if this is the wrong place to ask this, but are there special plans for students or educational purposes?

**Comment by izhangzhihao** [comment]
> I am sorry if this is the wrong place to ask this, but are there special plans for students or educational purposes?

Hi, I forget to enable the plan for students and teachers, now this plugin is free for students and teachers.

**Comment by kamronbatman** [comment]
@izhangzhihao, do you have any plans for specifically open source programmers? For example, Material UI provides free licenses to open source developers. Thanks!

I have been using your wonderful plugin for my project https://github.com/modernuo/modernuo. Unfortunately it looks like with Rider 2022.3 (for .NET 7), the C# features are no longer available.

**Comment by richardotomislav** [comment]
> > I am sorry if this is the wrong place to ask this, but are there special plans for students or educational purposes?
> 
> Hi, I forget to enable the plan for students and teachers, now this plugin is free for students and teachers.

I'm a student and the plugin is not detecting my student account. What do I need to do to register the plugin's license? 
Many thanks.

**Comment by izhangzhihao** [comment]
> I'm a student and the plugin is not detecting my student account. What do I need to do to register the plugin's license? Many thanks.

I don't know actually, maybe you need to submit a request for the JB team [here](https://www.jetbrains.com/community/education/#students)?

**Comment by richardotomislav** [comment]
> > I'm a student and the plugin is not detecting my student account. What do I need to do to register the plugin's license? Many thanks.
> 
> I don't know actually, maybe you need to submit a request for the JB team [here](https://www.jetbrains.com/community/education/#students)?

Sorry, maybe I wasn't clear enough. I have all JetBrains products with my student's license, but the plugin is not detecting this license. I'm attaching the window's dialog of the issue. 
In fact, I can open a new issue to make it more official and can be looked at in the future. 


![image](https://user-images.githubusercontent.com/113539857/201624534-334ecfe0-073a-40fc-bce3-7e4c18539e8a.png)

**Comment by lunananananananana** [comment]
Same here, also doesn't work for me

![image](https://user-images.githubusercontent.com/29987831/201688041-7c8f7170-3abd-490b-94bf-615a6abc593d.png)

![image](https://user-images.githubusercontent.com/29987831/201688838-43528820-20a3-433a-8d95-593055373ce8.png)

**Comment by kamronbatman** [comment]
For education licenses, you have to go through the process of trying to buy the license and then the price will change to zero.

**Comment by lunananananananana** [comment]
Oh I see, thanks. @izhangzhihao, you should mention somewhere that you need to do attempt to buy to get a student license

**Comment by wuxinhexa** [comment]
> For education licenses, you have to go through the process of trying to buy the license and then the price will change to zero.

hi kamronbatman, I want to get education licenses , but  when I try to buy the license, the price is not zero, do you know why?

**Comment by izhangzhihao** [comment]
Hi, sad news for students(It does NOT affect the license already obtained), I found that more than 65% of the orders were from students. But the basic free-to-use features are enough for students' coding, so I disabled the free educational licenses for students. If you really think that you need the paid features, please contact me by my email.

**Comment by wuxinhexa** [comment]
> Hi, sad news for students(It does NOT affect the license already obtained), I found that more than 65% of the orders were from students. But the basic free-to-use features are enough for students' coding, so I disabled the free educational licenses for students. If you really think that you need the paid features, please contact me by my email.

I see, thank you for your reply.

**Comment by ClumsyPenguin** [comment]
Hi, i was just wondering if you would enable a lifetime license for a higher price? I would be happy to pay more for a non-subscription price model.

**Comment by izhangzhihao** [comment]
In case of anyone wants a lifetime license, especially users using WeChat and Alipay, please read it [here](https://plugins.jetbrains.com/plugin/10080-rainbow-brackets/lifetime-license)

Hi @ClumsyPenguin, If you donate $50 [here](https://opencollective.com/intellij-rainbow-brackets/contribute/lifetime-license-user-49286/checkout) and email me, you will receive lifetime software license support. However, due to Jetbrains' restrictions, users must email me(izhangzhihao@hotmail.com) every year when the license expires and I will generate a new 100% discount coupon code for them to redeem for a one-year license. This way, you can receive our software support permanently and enjoy annual discounts.

**Comment by azeek** [comment]
All good. But why not provide a fallback license?

**Comment by Bleibtreu** [comment]
I don't think lifelong license should be provided, which consumes the passion and driving force of the author.

**Reply by izhangzhihao** [reply]
Thank you for considering from author's perspective. I also think about it from the end-user's side – some users strongly prefer a one-time payment, so I currently offer both options. And to be honest, users choosing the one-time purchase option aren't actually a significant portion. And this plugin will update regularly, for features and bug reports, please raise a new issue.
